cassovary-core/src/main/scala/com/twitter/cassovary/graph/tourist/PrevNbrCounter.scala

```diff
@@ -14,70 +14,70 @@
 package com.twitter.cassovary.graph.tourist
 
 import it.unimi.dsi.fastutil.ints._
-import java.util.Comparator
+import java.{util => jutil}
+import scala.collection.JavaConversions._
 
 /**
  * A NodeTourist that keeps track of the previous immediate neighbor of a
  * given node in visiting sequence.
  */
 class PrevNbrCounter(val numTopPathsPerNode: Option[Int], override val onlyOnce: Boolean)
-    extends InfoKeeper[Int, Int2IntMap, Int2ObjectMap[Int2IntMap]] {
+    extends InfoKeeper[Int2IntMap] {
 
   /**
    * Keep info only the first time a node is seen
    */
   def this() = this(None, false)
 
-  val infoPerNode = new Int2ObjectOpenHashMap[Int2IntOpenHashMap]
+  protected val underlyingMap = new Int2ObjectOpenHashMap[Int2IntOpenHashMap]
+
+  override def infoPerNode = underlyingMap.asInstanceOf[jutil.Map[Int, Int2IntMap]]
 
   /**
    * Priority queue and comparator for sorting prev nbrs. Reused across nodes.
    */
-   val comparator = new PrevNbrComparator(infoPerNode, true)
+   val comparator = new PrevNbrComparator(underlyingMap, true)
    val priQ = new IntHeapPriorityQueue(comparator)
 
+  override def recordInfo(id: Int, nodeMap: Int2IntMap) {
+    throw new UnsupportedOperationException("Use recordPreviousNeighbor instead")
+  }
+
   /**
-   * Record the previous neighbor {@code nodeId} of {@code id}.
+   * Record the previous neighbor `nodeId` of `id`
    */
-  def recordInfo(id: Int, nodeId: Int) {
+  def recordPreviousNeighbor(id: Int, nodeId: Int) {
     if (!(onlyOnce && infoPerNode.containsKey(id))) {
       nbrCountsPerNodeOrDefault(id).add(nodeId, 1)
```

Comment:
In current fastutil (6.4.4) in Build.scala there is no addTo. Let's make it in a separate PR, ok? [93]